Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -809,11 +809,27 @@ String WebExtension::bestMatchLocale()
     if (supportedLocales.size() == 1)
         return supportedLocales.first();
 
+    auto preferredLocale = defaultLanguage(ShouldMinimizeLanguages::No);
+
     bool exactMatch = false;
-    auto bestMatchIndex = indexOfBestMatchingLanguageInList(defaultLanguage(ShouldMinimizeLanguages::No), supportedLocales, exactMatch);
+    auto bestMatchIndex = indexOfBestMatchingLanguageInList(preferredLocale, supportedLocales, exactMatch);
     if (bestMatchIndex != notFound)
         return supportedLocales[bestMatchIndex];
 
+    auto preferredLocaleComponents = parseLocale(preferredLocale);
+
+    // The best match search above does not pick "zh" when the preferred locale is "zh-Hant",
```

Comment:
Why is that? Is it worth explaining that in this comment?

Reply:
I guess that would be in case there are other cases it would fail to match.

Reply:
I'll tweak the comment, but it is kind-of a black box to us on why.